src/reducers/profile-view.js

```diff
@@ -260,6 +261,26 @@ const viewOptionsPerThread: Reducer<ThreadViewOptionsPerThreads> = (
       return _updateThreadViewOptions(state, threadsKey, {
         selectedCallNodePath,
         expandedCallNodePaths,
+        selectedFunctionTableFunction: null,
+      });
+    }
+    case 'CHANGE_SELECTED_METHODTABLE_CALL_NODE': {
+      const { selectedFunction, threadsKey } = action;
+
+      const threadState = _getThreadViewOptions(state, threadsKey);
+
+      const previousSelectedFunction =
+        threadState.selectedFunctionTableFunction;
+
+      // If the selected function doesn't actually change, let's return the previous
+      // state to avoid rerenders.
+      if (selectedFunction === previousSelectedFunction) {
+        return state;
+      }
+
+      return _updateThreadViewOptions(state, threadsKey, {
+        selectedFunctionTableFunction: selectedFunction,
+        selectedCallNodePath: [],
```

Comment:
So you're resetting the selected call node path to `[]` whenever a selection in the function table is made. And then in some places this special `selectedCallNodePath` value causes the behavior to switch from "regular call tree" behavior to "function table" behavior.
I wonder if that's the right way to deal with this. I'd have expected one of these two options: 1. Either make the two mutually exclusive, so that only one of the properties is present at any given time, along with some discriminator, or 2. Make the two orthogonal to each other (i.e. remember the selected call node path even when the selected function changes). This solution seems somewhere in between.

Reply:
I like the last option more.